tests/test_literals.py

```diff
@@ -153,3 +153,25 @@ def test_parse_literal() -> None:
     assert literal.value == "value"
     assert literal.lang is None
     assert literal.datatype == "http://example.com/vocab#mytype"
+
+
+# def test_equality() -> None:
+#     """Test equality."""
+if True:  # pylint: disable=using-constant-test
+    from tripper import RDF, XSD, Literal
+
+    assert Literal("text", datatype=XSD.string) == "text"
+    assert Literal("text", lang="en") == "text"
+    assert Literal("text", lang="en") != Literal("text", lang="dk")
+    assert Literal("text") == "text"
+    assert Literal("text") != "text2"
+    assert Literal("text", datatype=RDF.HTML) != "text"
+    assert Literal(1) == 1
+    assert Literal(1) == 1
+    assert Literal(1) != 1.0
+    assert Literal(1) != "1"
+    assert Literal(1, datatype=XSD.double) == 1.0
```

Comment:
I meant like this (note: I have written the suggestion so that the test will pass, but still point out that the order of the arguments is important):
```suggestion
    assert Literal(1, datatype=XSD.double) == 1.0
    assert not 1.0 == Literal(1, datatype=XSD.double)
```
I believe that the expression in the second, reversed-order equality would evaluate to false, because the conversion of the non-`Literal` to `Literal` will now never take place. I may be wrong, but if I'm correct I think that this behavior needs to be pointed out.

Reply:
You are right, the reverse expression will evaluate to false. Do you think we should add tests like that?

Reply:
I see that you found out that newer versions of Python also allow reverting equality statement. Thanks!